libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -304,6 +364,32 @@ void HTTPClient::setTimeout(uint16_t timeout)
     }
 }
 
+bool HTTPClient::setRootCA(uint8_t * cert, size_t size)
+{
+    bool result = false;
+    if(_tcp && (_protocol == "https" || _port == 443)) {
+        auto client = static_cast<WiFiClientSecure*>(_tcp.get());
```

Comment:
If you follow the existing `TransportTraits` pattern, you can create a new TransportTraits class which performs root certificate verification instead of fingerprint verification. In `setRootCA` method, you can set `_transportTraits` to an instance of, say, `new TLSRootCertTraits(cert, size)`, which will then perform verification using the existing interface.